pkg/common/consumer/consumer_manager.go

```diff
@@ -120,6 +120,20 @@ type kafkaConsumerGroupManagerImpl struct {
 // Verify that the kafkaConsumerGroupManagerImpl satisfies the KafkaConsumerGroupManager interface
 var _ KafkaConsumerGroupManager = (*kafkaConsumerGroupManagerImpl)(nil)
 
+// ReconfigureError is a custom error type returned by the Reconfigure() function.
+type ReconfigureError struct {
+	MultiError error    // A MultiError with all ConsumerGroup failures.
+	GroupIds   []string // The ConsumerGroup IDs which failed to stop/start.
+}
+
+// Error implements the golang error interface.
+func (r *ReconfigureError) Error() string {
+	return r.MultiError.Error()
```

Comment:
Is this okay if r.MultiError is nil?  It won't be in the code that uses it but since this is an exported struct it might be good to check.

Reply:
No, good call - I was thinking of returning the GroupIDs in the error string anyway - should be made more robust - I'll fix it!